extension/src/popup/metrics/access.ts

```diff
@@ -14,47 +14,58 @@ import { registerHandler, emitMetric, MetricsData } from "helpers/metrics";
 import { METRICS_DATA } from "constants/localStorageTypes";
 import { AppState } from "popup/App";
 
-registerHandler<AppState>(grantAccess.fulfilled, () => {
-  emitMetric(METRIC_NAMES.grantAccessSuccess);
-});
-registerHandler<AppState>(rejectAccess.fulfilled, () => {
-  emitMetric(METRIC_NAMES.grantAccessFail);
-});
-registerHandler<AppState>(addToken.fulfilled, () => {
-  const metricsData: MetricsData = JSON.parse(
-    localStorage.getItem(METRICS_DATA) || "{}",
-  );
-  emitMetric(METRIC_NAMES.addToken, {
-    accountType: metricsData.accountType,
+// Defer registration to avoid circular dependency issues during module initialization
+// Register handlers after a microtask to ensure all modules are fully loaded
+Promise.resolve().then(() => {
```

Comment:
hm, how was this presenting itself when you ran your code? I haven't seen anyone else run into this issue, so I'm wondering if this is actually an issue in the codebase or with your setup